src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -610,6 +638,21 @@ private void DecodeWellKnownAttributeAppliedToMethod(ref DecodeWellKnownAttribut
             {
                 DecodeInterceptsLocationAttribute(arguments);
             }
+            else if (attribute.IsTargetAttribute(AttributeDescription.OverloadResolutionPriorityAttribute))
+            {
+                if (this.OverriddenMethod is not null)
+                {
+                    diagnostics.Add(ErrorCode.ERR_CannotApplyOverloadResolutionPriorityToOverride, arguments.AttributeSyntaxOpt);
+                }
+
+                MessageID.IDS_OverloadResolutionPriority.CheckFeatureAvailability(diagnostics, arguments.AttributeSyntaxOpt);
+
+                if (this is SourcePropertyAccessorSymbol or SourceEventAccessorSymbol)
+                {
+                    // Cannot put 'OverloadResolutionPriorityAttribute' on an accessor method.
+                    diagnostics.Add(ErrorCode.ERR_CannotApplyOverloadResolutionPriorityToAccessor, arguments.AttributeSyntaxOpt);
+                }
```

Comment:
>} [](http://example.com/codeflow?start=16&length=1)

Consider moving the `if (this.OverriddenMethod is not null)` case here and using `else if (this.OverriddenMethod is not null)` to avoid two errors for an overridden accessor.